controllers/cloud.redhat.com/pool_controller.go

```diff
@@ -133,6 +134,12 @@ func (r *NamespacePoolReconciler) getPoolStatus(ctx context.Context, pool crd.Na
 				case "error":
 					r.Log.Info("Error status for namespace. Deleting", "ns-name", ns.Name)
 					DeleteNamespace(ctx, r.Client, ns.Name)
+
+					r.Log.Info("Removing prometheus-operator associated with", "ns-name", ns.Name)
+					err := DeletePrometheusOperator(ctx, r.Client, ns.Name)
+					if err != nil {
+						r.Log.Error(err, fmt.Sprintf("Error deleting prometheus.%s", ns.Name))
```

Comment:
I don't think this is the right place to do this operation - the function is called `getPoolStatus` - this should not perform any kind of delete operation

Reply:
The reason I had put that there was because the function was already handling the deletion if `env-status` was set to `error`- Should I also move that out of `getPoolStatus`?

Reply:
Went ahead with a new approach and sent you a message in Slack with working output :)